CMakeLists.txt

```diff
@@ -0,0 +1,123 @@
+cmake_minimum_required(VERSION 3.27.6)
+
+project(cute_ext 
+  VERSION "1.1.0"
+  LANGUAGES CXX
+)
+
+if(NOT CMAKE_CXX_STANDARD)
+  set(CMAKE_CXX_STANDARD 17)
+  set(CMAKE_CXX_STANDARD_REQUIRED ON)
+endif()
+
+if(NOT DEFINED FETCHCONTENT_QUIET) 
+  set(FETCHCONTENT_QUIET OFF)
+endif()
+
+if(CMAKE_SOURCE_DIR STREQUAL "${CMAKE_CURRENT_SOURCE_DIR}")
```

Comment:
I think this is recipe for disaster for reusability scenario, I would not do this kind of thing. I know many project do, but I think it's just making it hard to apprehend and reuse a library. Because `CMAKE_SOURCE_DIR STREQUAL "${CMAKE_CURRENT_SOURCE_DIR}"` if you use it with something like HFC, it won't if you use it with something like plain FetchContent.

Reply:
I would just always think : configure the project for being reused by default, not for being maintained.